src/bespokelabs/curator/request_processor/config.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass, field
+from typing import Optional
+from litellm import get_supported_openai_params
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def _verify_non_negative(value: Optional[int], param_name: str) -> None:
+    """Helper function to verify that a value is non-negative if it exists.
+
+    Args:
+        value: The value to check
+        param_name: Name of the parameter for the error message
+
+    Raises:
+        ValueError: If the value is negative
+    """
+    if value is not None and value < 0:
+        raise ValueError(f"{param_name} must be >= 0")
```

Comment:
This is one place where pydantic is useful:
https://docs.pydantic.dev/latest/concepts/validators/